android-agent/src/test/java/io/opentelemetry/android/instrumentation/AndroidInstrumentationRegistryImplTest.kt

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.android.instrumentation
+
+import android.app.Application
+import io.mockk.mockk
+import io.opentelemetry.android.OpenTelemetryRum
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.fail
+
+class AndroidInstrumentationRegistryImplTest {
+    private lateinit var registry: AndroidInstrumentationRegistryImpl
+
+    @BeforeEach
+    fun setUp() {
+        registry = AndroidInstrumentationRegistryImpl()
+    }
+
+    @Test
+    fun `Find and register implementations available in the classpath when querying an instrumentation`() {
+        val instrumentation = registry.get(TestAndroidInstrumentation::class.java)
+
+        instrumentation?.install(mockk(), mockk())
```

Comment:
I'd avoid `?.` or assert that it's not null otherwise the test would be green because `instrumentation` is null.
it'd fail in L30 but because of another reason

Reply:
Got it, thank you. I think the `isTrue/isFalse` assertions should verify values not being null, however, I agree it's probably better to make sure. I've made the changes. Cheers!

Reply:
yep the next assertion would fail (correctly) but the reason would be wrong (because it's null and not because it returned the non-expected result).